Set nonce and add georchestra stylesheet to geoserver

f-necas opened the pull request:
This PR set nonce to geor-header tag (already set on script)

It also adds georchestraStylesheet to geoserver to allow customizations. 

Related to https://github.com/georchestra/header/pull/40

f-necas committed:
feat: set nonce and add georchestra styleheet to geoserver

landryb commented:
i think `georchestraStylesheet` should be rendered only if not empty, otherwise that generates a broken `<link>` element:
```
<link rel="stylesheet" type="text/css" href="../../../" nonce="">
```

landryb committed:
only render the <link> element if georchestraStylesheet is defined

landryb committed:
set a georchestra-header compatible CSP policy for wicket (#42)

now that we have a proper nonce set on the script/link dom elements, allow the
wicket web admin pages to:
- load the webcomponent header from self or the CDN
- set unsafe-eval for the webcomponent to work at all
- load css from self or googleapis (think web fonts)
- load fonts from self or gstatic (css loading fonts..)

for script-src, since https://cdn.jsdelivr.net/gh/georchestra/ ends with a
slash it will match all available versions of the header@v/header.js

cf georchestra/header#40 and previous wip in georchestra/georchestra#4403

landryb committed:
reenable strict CSP now that we have a nonce and a wide enough policy

reverts 6ca3b85833

landryb committed:
revert the other part of 6ca3b85 (eg REPORT_ONLY defaults to false)

landryb reviewed (approved):
tested locally with georchestra/header#40, i can reenable CSP on my testing instance and the header/css is properly loaded.

landryb committed:
drop 5px padding from header class to fix header height

the dummy header was readded in a1c4dc90 to fix tests, but it creates a 10px
height div on top of the geor-header, breaking the carefully crafted layout..
since geoserver.css was already tweaked in 5a18d3a we can tweak it even more.
